### PR TITLE
Use AWX_TASK_ENV when connecting to Red Hat services

### PR DESCRIPTION
**Pull request by wenottingham**
##### SUMMARY

I found this lovely hammer pattern in the notifications code and went looking for some nails.

c.f. https://github.com/ansible/awx/issues/3103

##### ISSUE TYPE
 - BugFix (?) Pull Request

##### COMPONENT NAME
 - API

### CODE REVIEW

**Review by ryanpetrello — APPROVED** [review]
Yea, this isn't *exactly* what this intended for, but I guess it's fairly reasonable.

File: awx/api/views/root.py [hunk 1]
```diff
@@ -203,10 +205,15 @@ def post(self, request):
...
-            if isinstance(exc, (ValueError, OSError)) and exc.args:
+            elif isinstance(exc, requests.exceptions.ProxyError):
+                msg = _("Unable to connect to proxy server.")
+            elif isinstance(exc, requests.exceptions.ConnectionError):
+                msg = _("Could not connect to subscription service.")
+            elif isinstance(exc, (ValueError, OSError)) and exc.args:
                 msg = exc.args[0]
-            logger.exception(smart_text(u"Invalid license submitted."),
-                             extra=dict(actor=request.user.username))
+            else:
```
Comment (ryanpetrello) [inline awx/api/views/root.py:214]: This wasn't in an `else` block before (it always ran).  Is this intentional?

Comment (wenottingham) [reply awx/api/views/root.py:214]: it's just *really* noisy:
```

2020-02-07 19:21:35,426 ERROR    awx.api.views.root Invalid license submitted.
Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/urllib3/connection.py", line 159, in _new_conn
    (self._dns_host, self.port), self.timeout, **extra_kw)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/urllib3/util/connection.py", line 80, in create_connection
    raise err
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/urllib3/util/connection.py", line 70, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/urllib3/connectionpool.py", line 594, in urlopen
    self._prepare_proxy(conn)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/urllib3/connectionpool.py", line 805, in _prepare_proxy
    conn.connect()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/urllib3/connection.py", line 301, in connect
    conn = self._new_conn()
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/urllib3/connection.py", line 168, in _new_conn
    self, "Failed to establish a new connection: %s" % e)
urllib3.exceptions.NewConnectionError: <urllib3.connection.VerifiedHTTPSConnection object at 0x7f80ae2df240>: Failed to establish a new connection: [Errno 111] Connection refused

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/requests/adapters.py", line 449, in send
    timeout=timeout
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/urllib3/connectionpool.py", line 638, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/urllib3/util/retry.py", line 399, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='subscription.rhsm.redhat.com', port=443): Max retries exceeded with url: /subscription/users/rhn-engineering-notting/owners (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f80ae2df240>: Failed to establish a new connection: [Errno 111] Connection refused',)))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/api/views/root.py", line 195, in post
    validated = get_licenser().validate_rh(user, pw)
  File "/__init__.py", line 241, in validate_rh
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/requests/api.py", line 75, in get
    return request('get', url, params=params, **kwargs)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/requests/api.py", line 60, in request
    return session.request(method=method, url=url, **kwargs)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/requests/adapters.py", line 510, in send
    raise ProxyError(e, request=request)
requests.exceptions.ProxyError: HTTPSConnectionPool(host='subscription.rhsm.redhat.com', port=443): Max retries exceeded with url: /subscription/users/rhn-engineering-notting/owners (Caused by ProxyError('Cannot connect to proxy.', NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f80ae2df240>: Failed to establish a new connection: [Errno 111] Connection refused',)))
2020-02-07 19:21:35,435 WARNING  awx.api.generics status 400 received by user admin attempting to access /api/v2/config/subscriptions/ from 192.168.124.1
2020-02-07 19:21:35,449 WARNING  django.request Bad Request: /api/v2/config/subscriptions/
```

Comment (ryanpetrello) [reply awx/api/views/root.py:214]: Okay, works for me 👍

**Review by ryanpetrello — APPROVED** [review]

**Pull request merged**